### ISSUE TITLE
class_row failing type checking in Pyright

### ISSUE THREAD

**Issue by troyswanson** [psycopg/psycopg#400]
The example code in the [Static Typing](https://www.psycopg.org/psycopg3/docs/advanced/typing.html) document is failing type checking in Pyright, the default type checker in Visual Studio Code.

<img width="658" alt="Screen Shot 2022-10-04 at 11 34 08 AM" src="https://user-images.githubusercontent.com/1420926/193875262-8fe4aacb-ee56-40cf-8f0c-9e88f22ae871.png">

The two errors that are being shown are...

Line 16:
```
Could not bind method "__enter__" because "Cursor[Person]" is not assignable to parameter "self"
  Type "Cursor[Person]" cannot be assigned to type "Cursor[Row@Cursor]"
    TypeVar "Row@Cursor" is covariant
      Type "Person" cannot be assigned to type "Row@Cursor" Pylance(reportGeneralTypeIssues)
```

Line 37:
```
Expression of type "Row@Cursor" cannot be assigned to return type "Person"
  "object*" is incompatible with "Person" Pylance(reportGeneralTypeIssues)
```

**Comment by dvarrazzo** [comment]
I think it's the same problem in #365 

We are about to release a new version including that bugfix.

Thank you!

**Comment by troyswanson** [comment]
Thank you! Apologies for not finding that dupe first.Sent from my iPhoneOn Oct 4, 2022, at 12:17 PM, Daniele Varrazzo ***@***.***> wrote:﻿
Closed #400 as completed.
